lib/newlib/stubs.c

```diff
@@ -0,0 +1,131 @@
+//! \file
+//! \brief Newlib stubs implementation.
+//! \details Not all newlib functionality is allowed (by default) in theCore.
+//! Examples include forking or using memory allocations.
+
+#include <platform/execution.h>
+
+#include <stdlib.h>
+#include <sys/unistd.h>
+#include <sys/stat.h>
+#include <sys/times.h>
+#include <errno.h>
+
+#undef errno
+extern int errno;
+
+void _exit(int status)
+{
+    (void)status;
+    ecl_abort();
+}
+
+int _close(int file)
+{
+    (void)file;
+    return -1;
+}
+
+char *__env[1] = { 0 };
+
+char **environ = __env;
+
+int _execve(const char *name, char * const *argv, char * const * env)
+{
+    (void)name;
+    (void)argv;
+    (void)env;
+    errno = ENOMEM;
+    return -1;
+}
+
+int _fork(void)
+{
+    errno = EAGAIN;
```

Comment:
How about returning `ENOSYS` (`man 2 fork` says `fork() is not supported on this platform (for example, hardware without a Memory-Management Unit).`)?